morpheus/messages/multi_message.py

```diff
@@ -269,24 +269,10 @@ def set_meta(self, columns: typing.Union[None, str, typing.List[str]], value):
 
                 # cudf is really bad at adding new columns
                 if (isinstance(df, cudf.DataFrame)):
-
-                    saved_index = None
-
-                    # Check to see if we can use slices
-                    if (not (df.index.is_unique and
-                             (df.index.is_monotonic_increasing or df.index.is_monotonic_decreasing))):
-                        # Save the index and reset
-                        saved_index = df.index
-
-                        df.reset_index(drop=True, inplace=True)
-
-                    # Perform the update via slices
+                    saved_index = df.index
+                    df.reset_index(drop=True, inplace=True)
```

Comment:
This needs further investigation. Because the only time you should need to reset the index would be with out of order indices. Its not clear why this fixes it.